### PR TITLE
Add RuboCop (and fix simple whitespace, magic comment, trailing comma offenses)

### PR DESCRIPTION
**Pull request by tagliala**
### Description
- Remove redundant magic comments
- Fix CR end of lines, trailing whitespaces, and other layout issues
- Add a default editorconfig to prevent trailing whitespaces and CRLF end of lines
- Add Rubocop and an action check as requested here: https://github.com/caxlsx/caxlsx/pull/187#issuecomment-1497131615

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [ ] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [ ] I added an entry to the [changelog](../blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by tagliala** [comment]
This change is only related to whitespaces and removal of magic comments to keep the PR as small as possible and prevent changes to production code

**Please consider merging** (maybe together with #178)

I suggest to review with a local branch and go commit by commit, or use `git diff --ignore-all-space`, or use the "hide whitespaces" functionality on GitHub

image

---

As I mentioned before, I'm available to temporarily be a maintainer. My main goal is to prepare the ground for #189 because #189 is important :)

/cc @johnnyshields @kiskoza

**Comment by johnnyshields** [comment]
I've read the entire diff line-by-line and this looks fine to merge.

**Comment by johnnyshields** [comment]
@kiskoza can you please consider making @tagliala a maintainer?

File: lib/axlsx/stylesheet/styles.rb [hunk 1]
```diff
@@ -444,10 +447,10 @@ def parse_border_options(options={})
           end
 
           border.prs << BorderPr.new({
-            :name => edge, 
-            :style => edge_b_opts[:style], 
-            :color => Color.new(:rgb => edge_b_opts[:color]) },
```
Comment (tagliala) [inline lib/axlsx/stylesheet/styles.rb:449]: I understand that the trailing comma at the end of array and hashes may be subject to preferences, but in the majority of cases the trailing comma is not there, so for uniformity they have been stripped out

**Comment by tagliala** [comment]
Hello, I've pushed some other commits to fix more offenses

With whitespaces hidden, changes to production code are pretty simple and should be reviewable in under 5 minutes

**Comment by kiskoza** [comment]
hi. I'm planning to review the changes today and I'll try to get write access or somebody who can merge it in. thanks for all the efforts, all the PRs are very helpful that you've created in the last few days

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+inherit_from: .rubocop_todo.yml
+
+AllCops:
```
Comment (kiskoza) [inline .rubocop.yml:3]: currently we only care about Layout and Style cops. would it make sense to disable all cops and enable only these namespaces (and maybe a few more that makes sense)? the todo would be much smaller and we're free to enable them again when we want to add further improvements to the codebase

e.g.:
```yml
AllCops:
  DisabledByDefault: true
  NewCops: enable
  SuggestExtensions: false
  TargetRubyVersion: 2.3

Layout:
  Enabled: true

Style:
  Enabled: true
```

**Review by kiskoza — APPROVED** [review]
I checked the code changes and all looks good to me. I left a few comments on improving configuration for rubocop, but this PR could be merged without resolving these comments

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -0,0 +1,1120 @@
+# This configuration was generated by
+# `rubocop --auto-gen-config`
```
[the remark this answers is not in the document]
Comment (tagliala) [reply .rubocop_todo.yml:2]: I 100% agree for the timestamps, I have some concerns with the offense counts. I was using that to decide whether or not an issue should be fixed, but I agree to have the no-offense-counts checked in the repo for the moment

**Pull request merged**